src/model/external_model.py

```diff
@@ -129,6 +129,8 @@ def parse_external_schedule(external_schedule):
     return {
         'repeatable': external_schedule.get('repeatable'),
         'start_datetime': external_schedule.get('startDatetime'),
+        'endOption': external_schedule.get('endOption'),
+        'endArg': external_schedule.get('endArg'),
```

Comment:
I would suggest using explicit names, e.g. `end_datetime` and `max_executions`, to make the logic more readable (in the files below).
Also, please use snake_case for naming in python

Reply:
I believe it would be more logical to have it set up this way because you can use the "end_option" to determine which case should be executed (in if statements and similar situations). For example, if we wanted to run something for a job that doesn't have an end date, we would need to check that both the "end_datetime" and "max_executions" are set to None. It's not a big problem when there are only two options, but if another option is added in the future, it would require more work and fields in ScheduleConfig class.

Reply:
:+1: ok, agreed :)